uvicorn/supervisors/watchgodreload.py

```diff
@@ -141,12 +148,10 @@ def __init__(
         for w in reload_dirs:
             self.watchers.append(CustomWatcher(w.resolve(), self.config))
 
-    def should_restart(self) -> bool:
+    def should_restart(self) -> Optional[List[Path]]:
         for watcher in self.watchers:
             change = watcher.check()
             if change != set():
-                message = "WatchGodReload detected file change in '%s'. Reloading..."
-                logger.warning(message, [c[1] for c in change])
-                return True
+                return list({Path(c[1]) for c in change})
```

Comment:
Isn't `change` already a set? Is it possible to have duplicated paths here?
```suggestion
                return [Path(c[1]) for c in change]
```

Reply:
This is required, you often get a "modified" and "added" or "deleted" in the same batch.